pynamodb/exceptions.py

```diff
@@ -95,14 +98,38 @@ def __init__(self, table_name):
         super(TableDoesNotExist, self).__init__(msg)
 
 
-class TransactWriteError(PynamoDBException):
+class TransactError(PynamoDBException):
+
+    def __init__(self, transact_items, *args, **kwargs):
+        super(TransactError, self).__init__(*args, **kwargs)
+        self._cancel_reasons = self.parse_cancel_reasons(transact_items=transact_items)
+
+    @property
+    def cancel_reasons(self):
+        return self._cancel_reasons
+
+    @staticmethod
+    def _get_reason_list_from_message(message):
+        reasons = TRANSACTION_CANCELED_REGEX.search(message)
```

Comment:
Rather confused about it. Is it a regex with a variable number of groups?

Reply:
no, should just be one :0

Reply:
ahh, it's indexing on a string

Reply:
```py
TRANSACTION_CANCELED_REGEX = re.compile(r'^.* \[(.*)\]$')
...
m = TRANSACTION_CANCELED_REGEX.match(message)
if not m:
   return None
return m.group(1).split(', ')
```

Reply:
We could also do something like:

```ipython
In [20]: re.compile(r'\[([A-Za-z]+)(?:,\s([A-Za-z]+))*\]').search(msg).groups()
Out[20]: ('None', 'ConditionalCheckFailed')
```

Reply:
Slightly less readable than a `split` IMO, but fine too -- maybe change `[A-Za-z]+` to `\w+` since we have no reason to assume the contrary?

Reply:
Possibly a matter of preference, but I find a single regex to be more readable than a regex + split combined parse